A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js

```diff
@@ -296,9 +283,26 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
         // switch page when desired page was changed, or new Failure detected
         if ((this.pageNameWhenUnselected != prevPage && this.currentPage == -1) || (this.PrevFailPage !== sFailPage)) {
             this.SwitchToPageName(this.LOWER_SCREEN_GROUP_NAME, this.pageNameWhenUnselected);
+        }
+
+        // ECAM all button
+        this.ecamAllButtonState = SimVar.GetSimVarValue("L:A32NX_ECAM_ALL_Push_IsDown", "Bool");
 
+        if (this.ecamAllButtonState == true && this.ecamAllButtonPrevState == false) { // button press
+            this.ecamCycleInterval = setInterval(() => {
```

Comment:
Another thing I'm not sure of is whether the first page switch should occur immediately or after one second

Reply:
Not sure about that one. I'll ask in the discord

Reply:
According to a pilot in the discord, it should also change immediately